config/cluster.yaml.template

```diff
@@ -0,0 +1,11 @@
+# base cluster settings
+base:
+    cluster_id: <id>
```

Comment:
i am of the opinion that you should not be able to specify cluster_id here (or at least not recommend setting it here)... 

Maybe, in the template, we can just comment it out and add a note saying "you can optionally choose to specify your cluster id here. However, please note that cluster ids need to be unique."

Reply:
Hrm... maybe... I'm not sure what the _right_ thing to do here is. It makes the interactive mode much more straight foward.

We spoke about this yesterday as well - do we need a .template file for this? It feels redundant if we ship a fully fleshed out file, which makes getting started far easier.

Reply:
This is similar to what do we do doAzureParallel with the generateConfigTemplate. We fill out a bunch of default values already. Actually, for doAzPar we should jsut ship a default template too :)

Reply:
i **think** i agree with this :P 

definitely should be using a .template for the spark config files (and maybe even the ssh.yaml one too)

Reply:
I think what I am saying to NOT have a .template for this one, just a cluster.yaml which looks like

```yaml
base:
    id: Spark
    vm_size: Standard_a2
    size: 0
    size_low_pri: 3
    ...
```

Reply:
agree - but without 'base'